pkg/scheduler/apis/config/validation/validation_test.go

```diff
@@ -575,3 +576,284 @@ func TestValidateKubeSchedulerConfigurationV1beta3(t *testing.T) {
+	}
+
+	extenderDuplicateBind := validConfig.DeepCopy()
+	extenderDuplicateBind.Extenders[0].BindVerb = "foo"
+	extenderDuplicateBind.Extenders = append(extenderDuplicateBind.Extenders, config.Extender{
+		PrioritizeVerb: "prioritize",
+		BindVerb:       "bar",
+	})
+
+	goodInvalidPlugins := validConfig.DeepCopy()
+	goodInvalidPlugins.Profiles[0].Plugins.Score.Enabled = append(goodInvalidPlugins.Profiles[0].Plugins.Score.Enabled, config.Plugin{Name: "PodTopologySpread", Weight: 2})
+
+	badInvalidPlugins := validConfig.DeepCopy()
+	badInvalidPlugins.Profiles[0].Plugins.Score.Enabled = append(badInvalidPlugins.Profiles[0].Plugins.Score.Enabled, config.Plugin{Name: "SelectorSpread"})
+
+	scenarios := map[string]struct {
+		expectedToFail bool
+		config         *config.KubeSchedulerConfiguration
+		errorString    string
+	}{
+		"good": {
+			expectedToFail: false,
+			config:         validConfig,
+		},
+		"bad-parallelism-invalid-value": {
+			expectedToFail: true,
+			config:         invalidParallelismValue,
+		},
+		"bad-resource-name-not-set": {
+			expectedToFail: true,
+			config:         resourceNameNotSet,
+		},
+		"bad-resource-namespace-not-set": {
+			expectedToFail: true,
+			config:         resourceNamespaceNotSet,
+		},
+		"non-empty-metrics-bind-addr": {
+			expectedToFail: true,
+			config:         metricsBindAddrInvalid,
+		},
+		"non-empty-healthz-bind-addr": {
+			expectedToFail: true,
+			config:         healthzBindAddrInvalid,
+		},
+		"bad-percentage-of-nodes-to-score": {
+			expectedToFail: true,
+			config:         percentageOfNodesToScore101,
+		},
+		"scheduler-name-not-set": {
+			expectedToFail: true,
+			config:         schedulerNameNotSet,
+		},
+		"repeated-scheduler-name": {
+			expectedToFail: true,
+			config:         repeatedSchedulerName,
+		},
+		"different-queue-sort": {
+			expectedToFail: true,
+			config:         differentQueueSort,
+		},
+		"one-empty-queue-sort": {
+			expectedToFail: true,
+			config:         oneEmptyQueueSort,
+		},
+		"extender-negative-weight": {
+			expectedToFail: true,
+			config:         extenderNegativeWeight,
+		},
+		"extender-duplicate-managed-resources": {
+			expectedToFail: true,
+			config:         extenderDuplicateManagedResource,
+		},
+		"extender-duplicate-bind": {
+			expectedToFail: true,
+			config:         extenderDuplicateBind,
+		},
+		"invalid-node-percentage": {
+			expectedToFail: true,
+			config:         invalidNodePercentage,
+		},
+		"invalid-plugin-args": {
+			expectedToFail: true,
+			config:         invalidPluginArgs,
+		},
+		"duplicated-plugin-config": {
+			expectedToFail: true,
+			config:         duplicatedPluginConfig,
+		},
+		"mismatch-queue-sort": {
+			expectedToFail: true,
+			config:         mismatchQueueSort,
+		},
+		"good-invalid-plugins": {
+			expectedToFail: false,
+			config:         goodInvalidPlugins,
+		},
+		"bad-invalid-plugins": {
+			expectedToFail: true,
+			config:         badInvalidPlugins,
```

Comment:
is there an errorString we should expect?

Reply:
Add `errorString` validation to all tests.